### DISCUSSION TITLE
Foreign key constraint name too long

### DISCUSSION THREAD

**Discussion by nick-rackauckas** [planetscale/discussion#569] [Bug Reports] [answered]
[Docs](https://planetscale.com/docs/concepts/foreign-key-constraints#foreign-key-constraint-names-change-on-every-deployment) say the name is generated each deployment, but that doesn't appear to be the case in the first deploymnet

Seeing the error `Constraint name <name> is longer than the 38 character limit (table: <table>)` when using MySql

Using with Prisma

Current workaround is to use Prisma's named constraint via map, as described [here](https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-3/named-constraints#option-1-i-want-to-maintain-my-existing-constraint-and-index-names)

Ex: `author     User?  @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")`

**Comment by tbarn** [comment]
Hey @nick-rackauckas! This is on our radar, I came across it with a different ORM and we will look into improving the linter on this. It looks like you found the workaround we would suggest for now! I'll update here when I have more to share.

**Reply by nick-rackauckas** [reply]
Thanks for the quick response!

**Reply by janpio** [reply]
Can you share the default migration Prisma generated when _not_ using `map`? That might surface the name that was considered too long by PlanetScale and might be useful as context.

**Reply by tbarn** [reply]
It should be good now, but in case any other auto generated name causes an issue, let us know!

**Comment by dbussink** [comment, marked as answer]
This should have been fixed now!
